v2/routes.go

```diff
@@ -352,18 +342,31 @@ func (s *Server) deleteSubscription(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) deleteAllSubscriptions(w http.ResponseWriter, _ *http.Request) {
-	size := len(s.pubSubAPI.GetSubscriptions())
-	if err := s.pubSubAPI.DeleteAllSubscriptions(); err != nil {
+	// update configMap
+	for _, subs := range s.subscriberAPI.SubscriberStore.Store {
+		cevent, _ := subs.CreateCloudEvents()
+		out := channel.DataChan{
+			ClientID: subs.GetClientID(),
+			Data:     cevent,
+			Status:   channel.DELETE,
+			Type:     channel.SUBSCRIBER,
+		}
+		s.dataOut <- &out
+	}
+
+	numSubDeleted, err := s.subscriberAPI.DeleteAllSubscriptions()
+
+	// Subscriptions could be partially deleted when there were errors
+	if numSubDeleted > 0 {
```

Comment:
This will be true when all subscriptions are deleted too right? numSubDeleted equals the number of subscriptions deleted.. you need a way to check whether the number of subscriptions before the delete == numSubDeleted? if not you can update the metrics with the difference?

Reply:
The metric simple pegs number of subs what were deleted, either all (err==nil) or some (when err!=nil).